airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx

```diff
@@ -19,29 +19,32 @@ const STYLES_BY_VARIANT: Readonly<Record<PillButtonVariant, string>> = {
 interface PillButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   active?: boolean;
   variant?: PillButtonVariant;
+  labels?: React.ReactNode[];
 }
 
-export const PillButton: React.FC<React.PropsWithChildren<PillButtonProps>> = ({
-  children,
-  active,
-  variant = "grey",
-  ...buttonProps
-}) => {
+export const PillButton: React.FC<PillButtonProps> = ({ active, variant = "grey", labels = [], ...buttonProps }) => {
   const buttonClassName = classNames(
     styles.button,
     {
       [styles.active]: active,
+      [styles.disabled]: buttonProps.disabled,
     },
-    buttonProps.disabled ? styles.disabled : STYLES_BY_VARIANT[variant],
+    STYLES_BY_VARIANT[variant],
     buttonProps.className
   );
-
   return (
     <button type="button" {...buttonProps} className={buttonClassName}>
-      <Text as="span" size="xs" className={styles.text}>
-        {children}
-      </Text>
-      <FontAwesomeIcon icon={faCaretDown} />
+      {labels.map((label, index) => (
+        <>
+          <div className={styles.labelContainer}>
+            <Text as="span" size="xs" className={styles.text}>
+              {label}
+            </Text>
+          </div>
+          {index !== labels?.length - 1 && <div className={styles.divider} />}
```

Comment:
Did you consider having a border-right style for each labelContainer instead of adding a div that represents the divider?

Reply:
I actually saw both variants: Tailwind makes the divider via css, and Bootstrap makes it via styling some empty tag.
Any benefits of making the divider via css only? 
Except, -1 empty node in the tree of course 😄